modules/heat_conduction/src/fvbcs/FunctorThermalResistanceBC.C

```diff
@@ -0,0 +1,177 @@
+    for (const auto & d : _dx)
+      _outer_radius += d;
+
+  // because the thermal conductivities are constant, we only need to compute
+  // the conduction resistance one time
+  computeConductionResistance();
+}
+
+void
+FunctorThermalResistanceBC::computeConductionResistance()
+{
+  Real r = _inner_radius;
+
+  for (std::size_t i = 0; i < _k.size(); ++i)
+  {
+    switch (_geometry)
+    {
+      case Moose::CoordinateSystemType::COORD_XYZ:
+        _conduction_resistance += _dx[i] / _k[i];
+        break;
+      case Moose::CoordinateSystemType::COORD_RZ:
+      {
+        _conduction_resistance += std::log((_dx[i] + r) / r) / _k[i];
+        r += _dx[i];
+        break;
+      }
+      default:
+        mooseError("Unhandled 'GeometryEnum' in 'FunctorThermalResistanceBC'!");
+    }
+  }
+}
+
+ADReal
+FunctorThermalResistanceBC::computeQpResidual()
+{
+  // radiation resistance has to be solved iteratively, since we don't know the
+  // surface temperature. We do know that the heat flux in the conduction layers
+  // must match the heat flux in the parallel convection-radiation segment. For a
+  // first guess, take the surface temperature as the average of _T and T_ambient.
+  _T_surface = 0.5 * (_T(&_face_info->elem()) + _T_ambient);
+
+  // total flux perpendicular to boundary
+  ADReal flux;
+
+  // resistance component representing the sum of the convection and radiation
+  // resistances, in parallel
+  computeParallelResistance();
+
+  // other iteration requirements
+  Real iteration = 0;
+  ADReal norm = 2 * _tolerance;
+  ADReal T_surface_previous;
+
+  // iterate to find the approximate surface temperature needed for evaluating the
+  // radiation resistance. We only do this iteration if we have radiation transfer.
+  if (_emissivity > 1e-8)
+    while (norm > (_tolerance * _alpha))
+    {
+      T_surface_previous = _T_surface;
+
+      // compute the flux based on the conduction part of the circuit
+      flux = (_T(&_face_info->elem()) - _T_surface) / _conduction_resistance;
+
+      computeParallelResistance();
+
+      // use the flux computed from the conduction half to update T_surface
+      _T_surface = flux * _parallel_resistance + _T_ambient;
+      _T_surface = _alpha * _T_surface + (1 - _alpha) * T_surface_previous;
+      norm = std::abs(_T_surface - T_surface_previous) / std::abs(T_surface_previous);
+
+      if (iteration == _max_iterations)
+      {
+        mooseWarning("Maximum number of iterations reached in 'FunctorThermalResistanceBC'!");
+        break;
+      }
+      else
+        iteration += 1;
+    }
+
+  // once we have determined T_surface, we can finally evaluate the complete
+  // resistance to find the overall heat flux. For Cartesian, dividing by the
+  // 'inner_radius' has no effect, but it is required for correct normalization
+  // for cylindrical geometries.
+  flux = (_T(&_face_info->elem()) - _T_ambient) / (_conduction_resistance + _parallel_resistance) /
+         _inner_radius;
```

Comment:
Does the non-functor version of this query at `_qp`? Use of `_face_info->elem()` would be problematic if this BC was on an internal boundary where the variable is subdomain restricted and only exists on the `_face_info->neighbor()` side

Reply:
You could query the face type, and set the elem to evaluate at based on that

Reply:
well this one very much has an outer boundary feel imo, since it s assuming radiation to infinity, an outer temperature etc

But the other BC I definitely need to change
I ll fix both just in case